### PR TITLE
Design change document node

### PR DESCRIPTION
**Pull request by asylves1**
# Description

Design Change

- Add notification progress bar to the document Node
- Add a thumbnail of the Pdf document

image

Testing:

- Start up local build
- Create new workflow and upload a new doc.
- The thumbnail should appear
- The progress bar should also appear after a moment when the extraction starts

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java [hunk 1]
```diff
@@ -362,16 +369,57 @@ private ResponseEntity<Void> uploadDocumentHelper(
...
+					g2d.dispose();
+
+					ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+					ImageIO.write(resizedBufferedImage, "png", outputStream);
+
+					byte[] thumbnailBytes = outputStream.toByteArray();
+					outputStream.close();
+
+					document.get().setThumbnail(thumbnailBytes);
+					pdfDocument.close();
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java:407]: There are a lot of calls to `close` various things in here that won't get triggered if there is an exception at some point in creating the thumbnail... consider using the `finally` block for these? Somethings will have to be re-scoped for this to work

```
try{
...
} catch (Exception e) {
...
} finally {
g2d.dispose()
outputStream.close()
...
}
```

**Review by dvince2 — APPROVED** [review]
This is really cool!

File: packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue [hunk 2]
```diff
@@ -5,7 +5,30 @@
 			<h6>
 				<span class="truncate-after-three-lines">{{ documentName }}</span>
 			</h6>
-			<tera-operator-placeholder :node="node" />
+			<tera-operator-placeholder v-if="!thumbnail" :node="node" />
+			<section v-else>
+				<img class="pdf-thumbnail" :src="thumbnail" alt="Pdf's first page" />
+			</section>
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue:11]: could this just be an img tag?

File: packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue [hunk 3]
```diff
@@ -128,6 +161,24 @@ watch(
...
 );
+
+watch(
+	() => document.value?.thumbnail,
+	() => {
+		if (document.value?.thumbnail?.length) {
+			thumbnail.value = `data:image/png;base64, ${document.value.thumbnail}`;
+		}
+	}
+);
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue:172]: i think you can just add this if statement to the above watcher and things will be the same

File: packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue [hunk 4]
```diff
@@ -138,4 +189,44 @@ watch(
...
+}
+
+.pdf-thumbnail {
+	padding-bottom: var(--gap-small);
+	border: 1px solid var(--surface-border-light);
+	border-radius: var(--border-radius-medium);
+}
+
+.done-container {
+	margin-top: 0.5rem;
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue:218]: use gap variables

File: packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue [hunk 5]
```diff
@@ -138,4 +189,44 @@ watch(
...
+}
+
+.action {
+	font-size: var(--font-caption);
+	color: var(--text-color-secondary);
+	text-align: center;
+}
+
+.pdf-thumbnail {
+	padding-bottom: var(--gap-small);
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/document/tera-document-operation-node.vue:212]: I think we want to use the gap variables with the numbers instead

**Review by shawnyama — APPROVED** [review]

**Pull request merged**